spec/support/link_check_matchers.rb

```diff
@@ -10,4 +10,14 @@
       raise "Failed when checking URL: \"#{check_link}\""
     end
   end
+end
+
+RSpec::Matchers.define :be_a_working_link do
+  match do |check_link|
+    begin
+      good_statuses.include? Faraday.head(check_link).status
+    rescue => e
+      raise "Failed when checking \"#{check_link}\": #{e.message}"
```

Comment:
What scenario does this help with? It can make some bugs harder to track down by throwing away the first stack trace. You could grab the first few lines and make them part of the message, but that might be overkill?